libnavigation-core/src/main/java/com/mapbox/navigation/core/replay/route/ReplayRouteMapper.kt

```diff
@@ -88,7 +87,17 @@ class ReplayRouteMapper @JvmOverloads constructor(
      * @return [ReplayEventBase] [List]
      */
     fun mapGeometry(geometry: String): List<ReplayEventBase> {
-        return replayRouteDriver.drivePointList(options, PolylineUtils.decode(geometry, 6))
+        return mapPointList(PolylineUtils.decode(geometry, 6))
+    }
+
+    /**
+     * Take a list of [Point] and map it to events that can be replayed by the [MapboxReplayer].
+     *
+     * @param points containing location coordinates to be replayed.
+     * @return [ReplayEventBase] [List]
+     */
+    fun mapPointList(points: List<Point>): List<ReplayEventBase> {
+        return replayRouteDriver.drivePointList(options, points)
```

Comment:
All of the functions in the `ReplayRouteMapper` return a `List`. And the replayer still takes events as a list
[`fun pushEvents(events: List<ReplayEventBase>)`](https://github.com/mapbox/mapbox-navigation-android/blob/03d1fc09fd954d38384b8ee8de2528ea74872cd7/libnavigation-core/src/main/java/com/mapbox/navigation/core/replay/MapboxReplayer.kt#L39)

So I don't think it is easier to convert back and forth at this time.

Reply:
The algorithms under the covers are still using lists, because they dedup locations on the route and they estimate the route curvatures. I agree that iterable sequence can be better, but in this case we're also sharing algorithms and moving code back and forth between mapbox java https://github.com/mapbox/mapbox-java/blob/1b867fe8e68b299f37e5591c8431df89881ca9c2/services-geojson/src/main/java/com/mapbox/geojson/utils/PolylineUtils.java#L39. Some of the decisions were based on the tools that exist in mapbox-java

It would be a big change to try and turn everything into a sequence

Reply:
Considering that we are trying to make replay "streamable", I also agree Sequence/Iterable/InputStream is where we should be going. But the `ReplayRouteMapper` is going to continue to be a `List` creator.

Take a look at `ReplayRouteDriver`, that is what we want to work on making into a `Sequence`. And then the mapper will convert the sequence into a `List` to maintain backwards compatibility.

I've been thinking about this too, so am just sharing the thoughts on this.

Reply:
That is essentially the approach I was taking here https://github.com/mapbox/mapbox-navigation-android/pull/6582

Instead of a Stream it's an `Iterator`. I'm not opposed to introducing it, but it still requires bug fixing
``` java
public interface ReplayEventStream extends Closeable, Iterator<ReplayEventBase>
```